File: Camera/Camera/AvatarImageView.m
//
//  AvatarImageView.m
//  Camera
//
//  Created by Paul Vagner on 10/23/15.
//  Copyright © 2015 Paul Vagner. All rights reserved.
//

#import "AvatarImageView.h"

@implementation AvatarImageView


- (void)layoutSubviews {
    //runs after the constraints have changed
    self.layer.cornerRadius = self.frame.size.width / 2;
    self.layer.masksToBounds = YES;
    
}

- (void)prepareForInterfaceBuilder {
    
    
    self.layer.cornerRadius = self.frame.size.width / 2;
    self.layer.masksToBounds = YES;
    
}

@end

File: Selfie/AvatarImageView.swift
//
//  AvatarImageView.swift
//  Selfie
//
//  Created by Paul Vagner on 10/26/15.
//  Copyright © 2015 Paul Vagner. All rights reserved.
//

import UIKit

@IBDesignable

class AvatarImageView: UIView {
    
    
    @IBInspectable var cornerRadius: CGFloat = 0 {
        
        didSet {
            
            layer.cornerRadius = cornerRadius
            layer.masksToBounds = cornerRadius > 0
            
        }
    }
    
    override func prepareForInterfaceBuilder() {
        
        layer.cornerRadius = self.frame.size.width / 2
        layer.masksToBounds = true
        
    }
    
    override func layoutSubviews() {
        
        layer.cornerRadius = self.frame.size.width / 2
        layer.masksToBounds = true
        
    }
    
}
